src/components/App/App.js

```diff
@@ -1,37 +1,75 @@
 import React, { useEffect, useState } from 'react';
 import { Route, Switch, useHistory } from 'react-router-dom';
 
+import Authentication from '../Authentication/Authentication.js';
+import { CurrentUserContext } from '../../contexts/CurrentUserContext.js';
 import Header from '../Header/Header.js';
-import Main from '../Main/Main.js';
 import Footer from '../Footer/Footer.js';
+import Main from '../Main/Main.js';
 import Register from '../Register/Register.js';
-import Authentication from '../Authentication/Authentication.js';
 import SavedNews from '../SavedNews/SavedNews.js';
 import SuccessPopup from '../SuccessPopup/SuccessPopup.js';
+import newsApi from '../../utils/NewsApi.js';
+import mainApi from '../../utils/MainApi.js';
 
 
-function App(props) {
-    const [isRegisterOpen, setRegisterOpen] = useState(false);
+function App() {
+    const [cards, setCards] = useState([]);
+    const [savedCards, setSavedCards] = useState([]);
+    const [keyword, setKeyword] = useState([]);
+
+    const [currentUser, setCurrentUser] = useState({});
+
     const [isAuthOpen, setAuthOpen] = useState(false);
+    const [isLoggedIn, setLoggedIn] = useState(false);
+    const [isRegisterOpen, setRegisterOpen] = useState(false);
     const [isNavMobileOpen, setNavMobileOpen] = useState(false);
-    const [isCardSaved, setCardSaved] = useState(false);
+    const [isNothingFoundOpen, setIsNothingFoundOpen] = useState(false);
     const [isSuccessPopupOpen, setSuccessPopupOpen] = useState(false);
+    const [isPreloaderOpen, setIsPreloaderOpen] = useState(false);
+
     const history = useHistory();
 
+    // global state for close on Esc
     useEffect(() => {
-        const close = (e) => {
-            if (e.keyCode === 27) {
-                closeAllPopups();
-            }
-        }
+        const close = (e) => { if (e.keyCode === 27) { closeAllPopups(); } }
```

Comment:
**might be better**

You shouldn't use `numbers` for checking what key has been pressed because such numbers don't tell other developers what kind of key it is. `(evt.key === “Escape”)`